### PR TITLE
Add exp validation and respond with a 401 when a token is expired

### PR DESCRIPTION
**Pull request by quantumew**
Related https://github.com/articulate/authentic-rb/pull/7

We had another task to ensure we were validating the exp claim so this adds that here as well. The JWT library I chose doesn't do any claim value validation, just ensures they are signed.

### CODE REVIEW

**Review by dbertram — APPROVED** [review]
This is the first `go` code I've ever reviewed, but it lgtm.

I'd suggest getting someone who knows `go` to review for style/convention though :)

**Review by chazu — APPROVED** [review]
Looks good to me - tests are green. :shipit:

File: main_test.go [hunk 1]
```diff
@@ -178,14 +211,20 @@ var _ = Describe("authentic", func() {
...
 			It("returns 401 response in Gin middleware", func() {
 				mockContext.Request.Header["Authorization"] = []string{"Bearer " + badToken}
-				authentic.NewMiddlewareCreator().CreateGinMiddleware()(mockContext)
+				middlewareCreator.CreateGinMiddleware()(mockContext)
 				json.NewDecoder(rec.Body).Decode(&body)
 				Expect(rec.Code).To(Equal(401))
 				Expect(body.Message).To(Equal("Unauthorized"))
 			})
 
+			It("returns 403 response in Gin middleware due to expired token", func() {
```
Comment (dbertram) [inline main_test.go]: 403 => 401 in description?

Comment (quantumew) [reply main_test.go]: Ah yea good catch e88b57937c832a793e0ee5307b9801af9d0cae90

**Comment by quantumew** [comment]
To clarify I misunderstood the RFC second hand. 401 on invalid/expired token (although distinguishing the two will provide some value as far as debugging goes), and a 403 when the token is good but the user does not have the necessary access level, scope/aud issue.

**Pull request merged**